regexec.c

```diff
@@ -248,19 +249,35 @@ S_regcppush(pTHX_ const regexp *rex, I32 parenfloor, U32 maxopenparen _pDEPTH)
                    (unsigned long)maxopenparen,
                    (long)parenfloor);
 
-    SSGROW(total_elems + REGCP_FRAME_ELEMS);
-
     DEBUG_BUFFERS_r(
         if ((int)maxopenparen > (int)parenfloor)
-            Perl_re_exec_indentf( aTHX_
+            Perl_re_exec_indentf(aTHX_
                 "rex=0x%" UVxf " offs=0x%" UVxf ": saving capture indices:\n",
                 depth,
                 PTR2UV(rex),
                 PTR2UV(rex->offs)
             );
     );
+
+    SSGROW(total_elems + REGCP_FRAME_ELEMS);
+
+    /* memcpy the offs inside the stack - it's faster than for loop */
     memcpy(&PL_savestack[PL_savestack_ix], rex->offs + parenfloor + 1, paren_elems_to_push * sizeof(*PL_savestack));
     PL_savestack_ix += paren_elems_to_push;
+
+    DEBUG_BUFFERS_r(
```

Comment:
I remove the nested DEBUG_BUFFERS_r - I didn't combine them though because I think the first is informing about the push operation and the second reporting its status.